File: 1496-path-crossing/1496-path-crossing.java
class Solution {
    public boolean isPathCrossing(String path) {
         
        Set<List<Integer>> set= new HashSet<>();
        int i = 0;
        int j = 0;

        set.add(Arrays.asList(i, j));

        for(char ch : path.toCharArray())
        {
            if(ch == 'N')
                j++;
            else if(ch == 'E')
                i++;
            else if(ch == 'S')
                j--;
            else
                i--;
            
            List<Integer> list = Arrays.asList(i, j);

            if(set.contains(list))
                return true;
            else
                set.add(list);
        }

        return false;
    }
}

File: 1496-path-crossing/1496-path-crossing.js
/**
 * @param {string} path
 * @return {boolean}
 */
var isPathCrossing = function(path) {
  let set = new Set();
  let i = 0;
  let j = 0;

  set.add(`${i},${j}`);

  for (let ch of path) {
    if (ch === 'N') j++;
    else if (ch === 'E') i++;
    else if (ch === 'S') j--;
    else i--;

    let position = `${i},${j}`;

    if (set.has(position)) {
      return true;
    } else {
      set.add(position);
    }
  }

  return false;
};